folding/src/standard_config.rs

```diff
@@ -0,0 +1,471 @@
+where
+    Self: 'static,
+    G: CommitmentCurve,
+    I: Instance<G> + Index<Chall, Output = G::ScalarField> + Clone,
+    W: Witness<G> + Clone,
+    W: Index<Col, Output = Vec<G::ScalarField>> + Index<Sel, Output = Vec<G::ScalarField>>,
+    Col: Hash + Eq + Debug + Clone + FoldingColumnTrait,
+    Sel: Ord + Copy + Hash + Debug,
+    Chall: Hash + Eq + Debug + Copy,
+    Str: Clone + Index<Col, Output = Vec<G::ScalarField>>,
+{
+    type Column = Col;
+
+    type Selector = Sel;
+
+    type Challenge = Chall;
+
+    type Curve = G;
+
+    type Srs = srs::SRS<G>;
+
+    type Instance = I;
+
+    type Witness = W;
+
+    type Structure = Str;
+
+    type Env = Env<G, Col, Chall, Sel, Str, I, W>;
+}
+///A generic Index based environment
+pub struct Env<G, Col, Chall, Sel, Str, I, W>
+where
+    G: CommitmentCurve,
+    I: Instance<G> + Index<Chall, Output = G::ScalarField> + Clone,
+    W: Witness<G> + Clone,
+    W: Index<Col, Output = Vec<G::ScalarField>> + Index<Sel, Output = Vec<G::ScalarField>>,
+    Col: Hash + Eq,
+{
+    instances: [I; 2],
+    witnesses: [W; 2],
+    next_evals: ColumnMemoizer<Col, G::ScalarField, 10>,
+    structure: Str,
+    _todo: PhantomData<(G, Col, Chall, Sel, Str)>,
+}
+
+//implementing FoldingEnv
+impl<G, Col, Chall, Sel, Str, I, W> FoldingEnv<G::ScalarField, I, W, Col, Chall, Sel>
+    for Env<G, Col, Chall, Sel, Str, I, W>
+where
+    G: CommitmentCurve,
+    I: Instance<G> + Index<Chall, Output = G::ScalarField> + Clone,
+    W: Witness<G> + Clone,
+    W: Index<Col, Output = Vec<G::ScalarField>> + Index<Sel, Output = Vec<G::ScalarField>>,
+    Col: FoldingColumnTrait + Eq + Hash,
+    Sel: Copy,
+    Str: Clone + Index<Col, Output = Vec<G::ScalarField>>,
+{
+    type Structure = Str;
+
+    fn new(structure: &Self::Structure, instances: [&I; 2], witnesses: [&W; 2]) -> Self {
+        // cloning for now, ideally should work with references, but that requires deeper
+        // refactorings of folding
+        let instances = instances.map(Clone::clone);
+        let witnesses = witnesses.map(Clone::clone);
+        let structure = structure.clone();
+        Self {
+            instances,
+            witnesses,
+            structure,
+            next_evals: ColumnMemoizer::new(),
+            _todo: PhantomData,
+        }
+    }
+
+    fn challenge(&self, challenge: Chall, side: Side) -> G::ScalarField {
+        let instance = match side {
+            Side::Left => &self.instances[0],
+            Side::Right => &self.instances[1],
+        };
+        // handled through Index in I
+        instance[challenge]
+    }
+
+    fn col(&self, col: Col, curr_or_next: CurrOrNext, side: Side) -> &Vec<G::ScalarField> {
+        let witness = match side {
+            Side::Left => &self.witnesses[0],
+            Side::Right => &self.witnesses[1],
+        };
+        // this should hold as long the Index implementations are consistent with the
+        // FoldingColumnTrait implementation.
```

Comment:
Feel free to use `FoldingColumnTrait` to link to the link, and detect old definitions while building the doc.

Reply:
Yes, you're right. I didn't know that rustdoc wasn't checking `//` :(.